poly_borders/borders_data.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include "poly_borders/help_structures.hpp"
+
+#include "base/control_flow.hpp"
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace poly_borders
+{
+class BordersData
+{
+public:
+  inline static double const kEqualityEpsilon = 1e-20;
+  inline static std::string const kBorderExtension = ".poly";
+
+  void Init(std::string const & bordersDir);
+  /// \brief Runs |MarkPoint(borderId, pointId)| for each borderId and it's pointId. See to
+  /// |MarkPoint| for more details.
+  void MarkPoints();
+
+  void RemoveEmptySpaceBetweenBorders();
+
+  void DumpPolyFiles(std::string const & targetDir);
+  Polygon const & GetBordersPolygonByName(std::string const & name) const;
+  void PrintDiff();
+
+private:
+  struct Processor
+  {
+    explicit Processor(BordersData & data) : m_data(data) {}
+    void operator()(size_t borderId);
+
+    BordersData & m_data;
+  };
+
+  /// \brief Some polygons could have sequentially same points - duplicates. Next method remove such
+  /// points and left only unique.
+  void RemoveDuplicatePoints();
+
+  /// \brief Iterates through all points of all polygons and finds equals point on the other
+  /// polygon. If such point finds method will create link "some border with id: anotherBorderId has
+  /// the same point with id: anotherPointId".
```

Comment:
please, reformat (parenthesize, and/or replace "id: " with "id=")

Reply:
```
  /// \brief Finds point on other polygons equals to point passed on argument. If such point is
  /// found, the method will create link ("some border with id = anotherBorderId has the same point
  /// with id = anotherPointId").
```
?